src/components/shared/modals/modalWithInput/index.tsx

```diff
@@ -0,0 +1,118 @@
+import React, {Dispatch, SetStateAction, useEffect} from 'react';
+import {Button, Modal, Text} from 'native-base';
+import Input from 'components/shared/input';
+import styles from './styles';
+import {useForm} from 'react-hook-form';
+import {schema} from './validations';
+import {yupResolver} from '@hookform/resolvers/yup';
+import {Keyboard} from 'react-native';
+
+interface Props {
+  buttonText: string;
+  headerText: string;
+  firstInputName: string;
+  isVisible: boolean;
+  setIsVisible: Dispatch<SetStateAction<boolean>>;
+  secondInputName?: string;
+  onSubmit: (data: {[x: string]: string}) => void;
+}
+
+const ModalWithInput = ({
+  headerText,
+  firstInputName,
+  secondInputName,
+  onSubmit,
+  isVisible,
+  setIsVisible,
+}: Props) => {
+  const defaultValues = {
+    [firstInputName]: '',
+    [secondInputName as string]: '',
+  };
+  const {
+    control,
+    formState: {errors},
+    setError,
+    clearErrors,
+    handleSubmit,
+    setValue,
+  } = useForm({
+    defaultValues: defaultValues,
+    resolver: yupResolver(schema),
+    mode: 'onSubmit',
+    reValidateMode: 'onBlur',
+  });
+
+  const onPress = (data: {[x: string]: string}) => {
+    if (!data.firstInputValue || !data.secondInputValue) {
```

Comment:
If the second field is optional, what happens when we only want to use one field? The form will always show an error, right?

Reply:
yes, implemented only one field after doing the logic, refactoring it now.